adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -0,0 +1,477 @@
+            )
+            self.dependency_graph.add_node(dependency_node)
+
+            # add a classifier if necessary
+            self.add_classifier(
+                _object, count, dependency_node, noun_lexicon_entry=noun_lexicon_entry
+            )
+
+            # TODO: handle X_IS_Y; not implemented in syntax yet either
+
+            # if colour is specified it, add it as an adjectival modifier
+            if IGNORE_COLORS not in self.situation.syntax_hints:
+                for property_ in _object.properties:
+                    if self.situation.ontology.is_subtype_of(property_, COLOR):
+                        color_lexicon_entry = self._unique_lexicon_entry(property_)
+                        color_node = DependencyTreeToken(
+                            color_lexicon_entry.base_form,
+                            color_lexicon_entry.part_of_speech,
+                            color_lexicon_entry.intrinsic_morphosyntactic_properties,
+                        )
+                        self.dependency_graph.add_edge(
+                            color_node, dependency_node, role=ADJECTIVAL_MODIFIER
+                        )
+
+            self.objects_to_dependency_nodes[_object] = dependency_node
+            return dependency_node
+
+        """Add a classifier to a given noun"""
+
+        def add_classifier(
+            self,
+            _object: SituationObject,
+            count: int,
+            noun_dependency_node: DependencyTreeToken,
+            *,
+            noun_lexicon_entry: LexiconEntry,
+        ) -> None:
+
+            # get the current possession relations so we can add wo de or ni de
+            possession_relations = [
+                relation
+                for relation in self.situation.always_relations
+                if relation.relation_type == HAS and relation.second_slot == _object
+            ]
+
+            # we can only handle one possession relation at a time right now
+            if len(possession_relations) > 1:
+                raise RuntimeError("Cannot handle multiple possession relations")
+            elif len(possession_relations) == 1:
+                # handle the possession relation if there is one. We don't need to case on person in Chinese
+                # since all possessives are expressed NP+de+NP
+                possessor = self._noun_for_object(possession_relations[0].first_slot)
+                de = DependencyTreeToken("de", PARTICLE)
+                self.dependency_graph.add_edge(de, possessor, role=CASE_POSSESSIVE)
+                self.dependency_graph.add_edge(
+                    possessor, noun_dependency_node, role=NOMINAL_MODIFIER_POSSESSIVE
+                )
+            # if the count is one, we're done since we're not using yi CLF currently
+            if count == 1:
+                return
+            #  https://github.com/isi-vista/adam/issues/782
```

Comment:
nit: If we are going to raise an exception, it's nice to put the issue in the exception message itself.  It saves the user who sees it some time.